File: src/main/java/io/github/dqualizer/dqlang/types/rqa/definition/stimulus/Workload.java
package io.github.dqualizer.dqlang.types.rqa.definition.stimulus;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.loadprofile.LoadProfile;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Workload {

    private WorkloadType type;

    @JsonProperty("load_profile")
    private LoadProfile loadProfile;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/rqa/definition/stimulus/Workload.kt
package io.github.dqualizer.dqlang.types.rqa.definition.stimulus

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.loadprofile.LoadProfile

data class Workload (
    val type: WorkloadType? = null,

    @JsonProperty("load_profile")
    val loadProfile: LoadProfile? = null
)

